### PR TITLE
Gruppe 3

### PR DESCRIPTION
**Pull request by kufuntu**

### CODE REVIEW

File: Java/src/main/java/com/gildedrose/Item.java [hunk 1]
```diff
@@ -1,21 +1,23 @@
 package com.gildedrose;
 
-public class Item {
+public abstract class Item {
```
Comment (judith600) [inline Java/src/main/java/com/gildedrose/Item.java:3]: according to our requirements the Item class must not be changed

File: Java/src/main/java/com/gildedrose/BaseItem.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.gildedrose;
+
+public abstract class BaseItem extends Item {
+
+    public BaseItem(String name, int sellIn, int quality) {
+        super(name, sellIn, quality);
+    }
+
+    @Override
+    public void updateQuality() {
```
Comment (judith600) [inline Java/src/main/java/com/gildedrose/BaseItem.java:10]: method update quality AND sellin, maybe updateItem is a better name

**Review by DGCAS20 — COMMENTED** [review]
Gute erweiterbare Architektur!

File: Java/src/main/java/com/gildedrose/SulfurasItem.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+public class SulfurasItem extends Item {
+    
+    public SulfurasItem(int sellIn, int quality) {
+        super("Sulfuras, Hand of Ragnaros", sellIn, quality);
+    }
+
+    @Override
+    public void updateQuality() {
+
```
Comment (judith600) [inline Java/src/main/java/com/gildedrose/SulfurasItem.java:11]: not finished or empty on purpose?

File: Java/src/main/java/com/gildedrose/AgedBrieItem.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package com.gildedrose;
+
+public class AgedBrieItem extends Item {
+
+    public AgedBrieItem(int sellIn, int quality) {
+        super("Aged Brie", sellIn, quality);
+    }
+
+    @Override
+    public void updateQuality() {
```
Comment (judith600) [inline Java/src/main/java/com/gildedrose/AgedBrieItem.java:10]: question: is the mix between data and logic in the item classes desired?

**Pull request merged**